pkg/deployment/deployment.go

```diff
@@ -67,7 +67,6 @@ func Deploy(
 	var deployFunc deployFuncDef
 	var deployName string
 	var deployLabel string
-	var foundService dataplanev1.OpenStackDataPlaneService
```

Comment:
foundService is declared with `:=` in the `range services` loop.

If I leave it in this PR, then I get the following error.
```
go vet ./...
# github.com/openstack-k8s-operators/dataplane-operator/pkg/deployment
pkg/deployment/deployment.go:70:6: foundService declared but not used
# github.com/openstack-k8s-operators/dataplane-operator/pkg/deployment
vet: pkg/deployment/deployment.go:70:6: foundService declared but not used
make: *** [Makefile:112: vet] Error 2
```

Reply:
it's b/c the "if !haveCephSecret {" block was also deleted, so foundService no longer needs to be declared outside the loop.